HaxeFlixel/flixel
# Add support for OpenFL 8

This is still a work-in-progress, but just like #2133 this adds support for OpenFL 8 via the new `drawQuads()` API. Instead of creating a compatibility layer and effectively reimplementing the Tilesheet API, it adds a `FlxDrawQuadsItem` that is more properly integrated into the current renderer.

<strike>Currently I'm in the process of getting [Travis builds](https://travis-ci.org/Gama11/flixel) green (I added jobs for OpenFL 8).</strike>

Work on #2068 might still continue after this - it's a full render refactor which adds some cool features. This PR really just adds OpenFL 8 compatibility, and due to `drawQuads()` this is possible much more quickly and with much fewer changes.

There will probably be a transition phase in which OpenFL 3.6.1 will still be supported for quite a while. This branch is compatible with OpenFL 3.6.1 Legacy, OpenFL 3.6.1 Next and OpenFL 8. This is a bit messy in places, but probably necessary.

### How to test this
- install my flixel branch via `haxelib git flixel https://github.com/Gama11/flixel draw-quads-2`
- install openfl dev via `haxelib git openfl https://github.com/openfl/openfl`
- for Lime, the latest Haxelib release (6.2.0) should be fine
- some demos and some flixel-addons classes need updates as well, there's branches here: [[1]](https://github.com/gama11/flixel-demos/tree/draw-quads), [[2]](https://github.com/gama11/flixel-addons/tree/draw-quads)

### Limitations
- Shader filters (per-game / per-camera, such as in the Filters demo) are not supported yet. @jgranick is working on the OpenFL side of this.
- Per-sprite filters already work! However, they need to be adjusted a bit (see FlxBunnyMark).
- `flixel.effects.postprocess` is currently not supported, and might be deprecated since you can do basically the same thing with filters.
- Blend modes are not supported (lack of support in `drawQuads()`)
- `FlxPreloader` currently falls back to OpenFL's default preloader

Feedback is appreciated!
